px4_ros2_cpp/test/integration/global_navigation.cpp

```diff
@@ -0,0 +1,112 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+/**
+ * Summary:
+ *  Tests that measurements sent via the interface are fused into PX4.
+ * - `GlobalPositionInterfaceTest` fixture:
+ *   - Test Cases:
+ *     1. FuseAll: Sends measurements with all fields set. Expects cs_aux_gpos to be set to true.
+ */
+
+
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+#include <px4_msgs/msg/estimator_status_flags.hpp>
+#include <px4_ros2/navigation/experimental/global_position_measurement_interface.hpp>
+#include "util.hpp"
+
+using namespace std::chrono_literals;
+using px4_ros2::GlobalPositionMeasurement, px4_ros2::GlobalPositionMeasurementInterface;
+using px4_msgs::msg::EstimatorStatusFlags;
+
+class GlobalPositionInterfaceTest : public BaseTest
+{
+protected:
+  void SetUp() override
+  {
+    _node = initNode();
+
+    _global_navigation_interface = std::make_shared<GlobalPositionMeasurementInterface>(*_node);
+    ASSERT_TRUE(_global_navigation_interface->doRegister()) <<
+      "Failed to register GlobalPositionMeasurementInterface.";
+
+    // Subscribe to PX4 EKF estimator status flags
+    _subscriber = _node->create_subscription<EstimatorStatusFlags>(
+      "/fmu/out/estimator_status_flags", rclcpp::QoS(10).best_effort(),
+      [this](EstimatorStatusFlags::UniquePtr msg) {
+        _estimator_status_flags = std::move(msg);
+      });
+  }
+
+  void resetFlags()
+  {
+    // Wait for PX4 to stop fusing external measurements
+    while (_estimator_status_flags->cs_aux_gpos) {
+      rclcpp::spin_some(_node);
+      std::this_thread::sleep_for(_k_sleep_interval);
+    }
+
+    _estimator_status_flags.reset();
+  }
+
+  void waitForMeasurementUpdate(
+    std::unique_ptr<GlobalPositionMeasurement> measurement,
+    const std::function<bool(const EstimatorStatusFlags::UniquePtr &)> & is_fused_getter
+  )
+  {
+    auto start_time = std::chrono::steady_clock::now();
+
+    // Wait for PX4 estimator flags to confirm proper measurement fusion
+    while (!_estimator_status_flags || !is_fused_getter(_estimator_status_flags)) {
+
+      // Send measurement
+      measurement->timestamp_sample = _node->get_clock()->now();
+      ASSERT_NO_THROW(
+        _global_navigation_interface->update(*measurement)
+      ) << "Failed to send position measurement update via GlobalPositionMeasurementInterface.";
+
+      rclcpp::spin_some(_node);
+
+      const auto current_time = std::chrono::steady_clock::now();
+      const auto elapsed_time = std::chrono::duration_cast<std::chrono::milliseconds>(
```

Comment:
Is the cast required here?
```suggestion
      const auto elapsed_time = current_time - start_time;
```